### PR TITLE
Adds component for admin to select courses to generate codes for

### LINKED ISSUE #87: Add Create CodeCsv component
The code csv create component should include fields for: name, number of codes, and courses you want to show. You'll need to include a request to grab all the courses so that the admin can pick which course they want to add.

### PR DESCRIPTION
**Pull request by lbkchen**
Resolves #87

### CODE REVIEW

**Comment by lbkchen** [comment]
screen shot 2016-11-02 at 10 21 09 pm

File: app/assets/javascripts/admins/app.js [hunk 1]
```diff
@@ -6,11 +6,13 @@ import { ReactRoutes } from '../shared/routes'
 
 import AdminDashboardPage from './components/AdminDashboardPage'
 import UpdateAdminPage from './components/UpdateAdminPage'
-
+import GenerateCodeCsvModal from './components/GenerateCodeCsvModal'
+console.log(ReactRoutes.codeCsvListPath())
```
Comment (charlesx2013) [inline app/assets/javascripts/admins/app.js]: remove pls

File: app/assets/javascripts/admins/components/GenerateCodeCsvModal.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      this.setState( { courses: response.courses })
+    }, (error) => {
+      console.log('error')
+    })
+  }
+
+  updateCourseList(course) {
+    // Removes `course` if already in active course list
+    // Otherwise adds `course` to active course list
+    var index = this.state.active_courses.indexOf(course)
```
Comment (charlesx2013) [inline app/assets/javascripts/admins/components/GenerateCodeCsvModal.js]: try using a set instead with ids

File: app/assets/javascripts/admins/components/GenerateCodeCsvModal.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    if (index >= 0) {
+      this.state.active_courses.splice(index, 1)
+    } else {
+      this.state.active_courses.push(course)
+    }
+    console.log(this.state.active_courses)
```
Comment (charlesx2013) [inline app/assets/javascripts/admins/components/GenerateCodeCsvModal.js]: remove!

File: app/assets/javascripts/shared/routes.js [hunk 4]
```diff
@@ -67,6 +68,9 @@ class APIRoutes {
   static editSubsectionPath(id)  { return APIRoutes.createRoute(`admins/subsections/${id}`)}
   static editSectionPath(id)     { return APIRoutes.createRoute(`admins/sections/${id}`)}
   static editCoursePath(id)      { return APIRoutes.createRoute(`admins/courses/${id}`)}
+
+  // Codes
```
Comment (charlesx2013) [inline app/assets/javascripts/shared/routes.js]: remove

**Comment by lbkchen** [comment]
Should be gucci now?

**Comment by charlesx2013** [comment]
👍

**Pull request merged**